### PR TITLE
feat(fs): Uploads task for mass-upload

### LINKED ISSUE #61: Add the possibility to add mass upload
### Description
Currently Kestra allows for uploading files via SFTP/FTP/... , but it only allows to upload a single file per connection
Besides the connection only supporting a single upload, it is also required to specify the filename on both from and to locations. 
In case we want to upload multiple files we'll have to go over each file individually.

### Solution
Have the option to provide the _from_ as a folder or have the option to specify an array of files, the _to_ should in both cases point to a directory. 

**Upload the complete folder**
`- id: "uploadDir"
                type: "io.kestra.plugin.fs.sftp.Upload"
                host: ftp.server.com
                port: 22
                username: user
                password: secretpassword
                from: "{{outputs.generateArray.**outputDir**}}"
                to: /ftpserver/output/`

**Upload an array of files**
`- id: "uploadArray"
                type: "io.kestra.plugin.fs.sftp.Upload"
                host: ftp.server.com
                port: 22
                username: user
                password: secretpassword
                from: "{{outputs.generateArray.**outputFiles**}}"
                to: /ftpserver/output/`

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes #61

### CODE REVIEW

File: src/main/java/io/kestra/plugin/fs/ftp/Uploads.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    protected String proxyPort;
+    protected Proxy.Type proxyType;
+    @Builder.Default
+    protected Boolean rootDir = true;
+    @Builder.Default
+    protected String port = "21";
+    @Builder.Default
+    protected Boolean passiveMode = true;
+    @Builder.Default
+    protected Boolean remoteIpVerification = true;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/fs/ftp/Uploads.java:50]: I think all those can be private as we use builders all over the place.
The same apply to all other tasks.

File: src/main/java/io/kestra/plugin/fs/vfs/Uploads.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+                title = "The fully-qualified URIs that point to source data mapped to its fully-qualified destination URI"
+        )
+        Map<URI, URI> files;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/fs/vfs/Uploads.java:66]:
```suggestion
        private Map<URI, URI> files;
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]
Some minor code feedback, not mandatory.
But it misses an Uploads task for SFTP.

File: src/main/java/io/kestra/plugin/fs/sftp/Uploads.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    protected String passphrase;
+    protected String proxyHost;
+    protected String proxyPort;
+    protected String proxyUser;
+    protected String proxyPassword;
+    protected String proxyType;
+    @Builder.Default
+    protected Boolean rootDir = true;
+    @Builder.Default
+    protected String port = "22";
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/fs/sftp/Uploads.java:49]: Can be private

File: src/main/java/io/kestra/plugin/fs/vfs/Uploads.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class Uploads extends AbstractVfsTask implements RunnableTask<Uploads.Output> {
+    @PluginProperty(dynamic = true)
+    @Schema(
+            title = "The files to copy, must be internal storage URIs"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/fs/vfs/Uploads.java:27]:
```suggestion
            title = "The files to upload, must be internal storage URIs"
```

File: src/main/java/io/kestra/plugin/fs/vfs/Uploads.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+                title = "The fully-qualified URIs that point to source data mapped to its fully-qualified destination URI"
+        )
+        private Map<URI, URI> files;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/fs/vfs/Uploads.java:66]: I wonder if just the list wouln't be better, I'm not sure we properly render URI as map key in the UI

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/fs/vfs/Uploads.java (hunk 5 above)
Comment (brian-mulier-p) [reply src/main/java/io/kestra/plugin/fs/vfs/Uploads.java:66]: :+1: I'll go with that

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**